### ISSUE TITLE
Using MarkupEditor in a scrollview with other ui components

### ISSUE THREAD

**Issue by JohnKuan** [stevengharris/MarkupEditor#213]
Hi @stevengharris, would like to know if you have considered whether MarkupEditorView could be in a Scrollview, along side with other UI components such as a TextField etc, in the ScrollView.

In my example project, I am using it with a couple of TextFields and the MarkupEditorView is the last component at the bottom. I am aware that in that scenario, it should require turning off the scrolling on the webView of MarkupEditorView, but another concern will be to determine the intrinsic size of the webView.

**Comment by stevengharris** [comment]
No, I haven't thought about it directly. I was using many MarkupWKWebViews in a List for quite a while, which worked okay from a scrolling and even view re-use perspective, but ultimately I replaced this with the approach in  https://github.com/stevengharris/MarkupEditor/discussions/178. I'm using this approach to put many contentEditable divs along with non-editable ones in a single MarkupWKWebview so that I can scroll the overall page and let the web view machinery deal with the sizing/flow. I'm not sure if this kind of approach would work for you, since the items in the web view are all HTML entities, not iOS views, but it might be something to consider.

I do think the only times I've done something similar to what you're talking about is when the MarkupWKWebView itself is non-scrolling by autosizing. Since I'm not using a list of autosized views any more, I'd have to go digging a bit to reconstruct exactly how I dealt with it. I believe I just used the `MarkupDelegate.markup(_:heightDidChange:)` callback to adjust the frame size.

**Comment by ranbeuer** [comment]
Hi, @stevengharris  I'm trying to use the approach you mentioned in the last part of your response: using `MarkupDelegate.markup(_:heightDidChange:)` in order to update the ui, and make the control's height dynamic while being in a ScrollView, but it only updates a couple of times. I tried setting this up along with it `webView?.scrollView.isScrollEnabled = false` to see if it would work, but even though I type Enter while typing, and enter several lines, the method is not called when the text starts growing.
Am I missing something on how to continuously detect the change of the content's height? 
Thanks!

**Comment by stevengharris** [comment]
I will have to take a look at it, but I think I might have eliminated some of the heightDidChange callbacks when IIRC I discovered that the input callbacks (i.e., some input happened on the JavaScript side) resulted in the same Swift-side code being executed twice. But I might have messed something up, no question. I'll try to come up with a simple example of auto-side-adjustment unless you already have one. What I have in mind is just a reworked version of the demo app that adjusts the WKWebView size as you type into it.

**Comment by ranbeuer** [comment]
This is my current example I have:
```
struct MarkupEditorExampleView: View, MarkupDelegate {
    
    let defaultHeight = 50.0
    @State var htmlString: String = ""
    @State var webView: MarkupWKWebView?
    @State var height: CGFloat = 50.0
    
    var body: some View {
        ScrollView(.vertical) {
            VStack {              
                    MarkupEditorView(markupDelegate: self, html: $htmlString)
                        .frame(height: height)                
            }
            .padding()
        }
    }

    func markupInput(_ view: MarkupWKWebView) {
        webView = view
        webView?.scrollView.isScrollEnabled = false
    }
    
    func markup(_ view: MarkupWKWebView, heightDidChange newHeight: Int) {
        if newHeight.toCgFloat() + defaultHeight  != self.height {
            self.height = defaultHeight + newHeight.toCgFloat()
        }
        print(self.height)
    }
}
```

Please, tell me if I'm missing something.

**Comment by ranbeuer** [comment]
I have one more question, say I need two or more markup editors in a single view, I wrote a wrapper view for it so that I don't have them sharing delegate, so, my question is, is there a way for them to not share the webview? so that Each MarkupEditorView behaves independently of the ones inside other wrapper views.
